eng/sdk/sync_sdk.py

```diff
@@ -44,26 +47,45 @@ def update_emitter(version: str):
     logging.info('Update emitter-package.json to use @azure-tools/typespec-java version %s', version)
 
 
+def get_generated_folder_from_artifact(module_path: str, type: str, artifact: str) -> str:
+    path = os.path.join(module_path, 'src', type, 'java', 'com')
+    for seg in artifact.split('-'):
+        path = os.path.join(path, seg)
+    path = os.path.join(path, 'generated')
+    return path
+
+
 def update_sdks():
     for tsp_location_file in glob.glob(os.path.join(sdk_root, 'sdk/*/*/tsp-location.yaml')):
-        artifact_path = os.path.dirname(tsp_location_file)
-        artifact = os.path.basename(artifact_path)
+        tsp_location_path = os.path.dirname(tsp_location_file)
+        artifact = os.path.basename(tsp_location_path)
 
         if artifact in skip_artifacts:
             continue
 
+        module_path = os.path.join(sdk_root, os.path.dirname(tsp_location_path))
+        generated_samples_path = get_generated_folder_from_artifact(module_path, 'samples', artifact)
+        generated_test_path = get_generated_folder_from_artifact(module_path, 'test', artifact)
+        generated_samples_exists = os.path.isdir(generated_samples_path)
+        generated_test_exists = os.path.isdir(generated_test_path)
+
         logging.info('Generate for module %s', artifact)
 
         cmd = [
             'pwsh',
             os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Sync.ps1'),
-            artifact_path
+            tsp_location_path
         ]
         subprocess.check_call(cmd, cwd=sdk_root)
 
         cmd[1] = os.path.join(sdk_root, 'eng/common/scripts/TypeSpec-Project-Generate.ps1')
         subprocess.check_call(cmd, cwd=sdk_root)
 
+        if not generated_samples_exists:
```

Comment:
What is the case where `generated folder` not in sdk repo?

Reply:
service can generate it, copy some, then delete the "generated" folder.

in some case, some string in there would trigger cspell.

Reply:
But when service generate next api-version, regen these test/sample still helpful.

Reply:
I see. Thanks